### ISSUE TITLE
kde session restore makes goldendict lookup hash number on login 

### ISSUE THREAD

**Issue by Konfekt** [goldendict/goldendict#1165]
As discussed at https://github.com/goldendict/goldendict/issues/1041 and hinted at by the reply https://askubuntu.com/a/1096634/364843 `KDE  Plasma 5.12`'s session server `ksmserver` makes `GoldenDict  1.5.0-RC2+git` look up a long hash number on the user's login.

**Comment by Konfekt** [comment]
This hash number is passed to `goldendict` by the parameter `-session`.
For example, `ksmserver` calls `goldendict` with the parameters
```
restartCommand6=goldendict,-session,1018b17e167160000148155146100000028080013_1482152563_268421
```
Apparently `GoldenDict` does not accept the parameter `-session`. 

Is this against the `Qt` application standard, or should `ksmserver` take this into account and not try to connect `GoldeDict` to a session on session restore?

**Comment by Konfekt** [comment]
Perhaps GoldenDict should best simply bail out when passing parameters starting with a dash, or at least for `-session`.

**Comment by vedgy** [comment]
Are you testing the latest Goldendict git version or a version based on an older commit? Make sure that your version is based on d65a1f7869e83eba5c01617253c46ac7f63e3514 or later commit. Starting from the parent of this commit [`QSessionManager::RestartNever`](https://doc.qt.io/qt-5/qsessionmanager.html#RestartHint-enum) hint is passed to the session manager, which should fix your issue. The reporter of #1041 has confirmed that the fix works. Isn't this issue a duplicate of the already fixed and closed #1041?

**Comment by Konfekt** [comment]
It is version `GoldenDict 1.5.0-RC2`. I am sorry, I did miss that session handling was fixed but thought the opener fixed it by other means.
